### PR TITLE
Mb dev

### PR DESCRIPTION
**Pull request by mboissel**
main change : add bcftool cmd to filter chr at the vcf level
fix merge list vcf problem with big list : loop on batch
fix sort always needed for vcf
add option to not remove tmp folder

### CODE REVIEW

File: R/check_genotype.R [hunk 1]
```diff
@@ -108,44 +113,123 @@ create_genotype_matrix <- function(
...
-        if (!is_gzvcf) {
+        # if (!is_gzvcf) {
           system(paste(
             bin_path[["bcftools"]], "sort --output-type z --output-file", iout, ivcf
           ))
-        } else {
-          system(paste("cp", ivcf, iout))
-        }
+        # } else {
+          # system(paste("cp", ivcf, iout))
```
Comment (mcanouil) [inline R/check_genotype.R]: Use file.copy

File: R/check_genotype.R [hunk 2]
```diff
@@ -108,44 +113,123 @@ create_genotype_matrix <- function(
...
+    message(paste("Go for", nb_batch, "batchs"))
+    indice <- c(seq(1, length(vcfs_to_merge), by = 2000), length(vcfs_to_merge))
+    for (i in 1:nb_batch) {
+      message(">>> ", i)
+      indice_batchi <- (indice[i]+1):indice[i+1] ## (start+1:end) otherwise the final merge have borne duplicated
+      if(i==1) {indice_batchi <- indice[i]:indice[i+1]}
+
+      data.table::fwrite(
+        x = data.table::data.table(vcfs_to_merge[indice_batchi]),
+        file = file.path(output_tmp_dir, paste0("list_vcf",i,".txt")),
```
Comment (mcanouil) [inline R/check_genotype.R]: Spaces after commas

File: R/check_genotype.R [hunk 3]
```diff
@@ -108,44 +113,123 @@ create_genotype_matrix <- function(
...
+      merged_vcf_batchi <- file.path(output_tmp_dir, paste0(output_name, "batch",i,".vcf.gz"))
+      system(paste(
+        bin_path[["bcftools"]],
+        "merge -m none --file-list", file.path(output_tmp_dir, paste0("list_vcf",i,".txt")),
+        "-O z >", merged_vcf_batchi
+      ))
+      ## indexing vcf
+      system(paste(bin_path[["tabix"]], "-f -p vcf", merged_vcf_batchi))
+    }
+    if(nb_batch==1) {
```
Comment (mcanouil) [inline R/check_genotype.R]: Spaces

File: R/check_genotype.R [hunk 4]
```diff
@@ -185,8 +269,7 @@ create_genotype_matrix <- function(
     file = file.path(output_directory, paste0(output_name, ".tsv.gz")),
     col.names = TRUE
   )
-
-  unlink(output_tmp_dir, recursive = TRUE)
+  if(clean_tmp_folder) {unlink(output_tmp_dir, recursive = TRUE)}
```
Comment (mcanouil) [inline R/check_genotype.R]: Spaces and no braces

File: R/compress_coverage.R [hunk 5]
```diff
@@ -34,7 +34,7 @@ compress_coverage <- function(
   bin_path = list(
     awk = "/usr/bin/awk",
     bgzip = "/usr/bin/bgzip",
-    bgzip2 = NULL
+    bgzip2 = "bzip2"
```
Comment (mcanouil) [inline R/compress_coverage.R]: (b)gzip is not bzip2.
Use bzfile and related function to handle bzip2 compressed files

**Review by mcanouil — CHANGES_REQUESTED** [review]
Please fix the comments and ensure code style is similar to the overall code, _i.e._, [tidyverse style guide](https://style.tidyverse.org/syntax.html#spacing) (mostly).
Call to `system()` should be avoided as much as possible, especially when there is an R API/function.
Also, any additional system dependencies should be listed in `DESCRIPTION`.

**Comment by mcanouil** [comment]
@mboissel Any news on this?

**Comment by mboissel** [comment]
> @mboissel Any news on this?

Yes I've taken into account your several comments (indent, style and binary to use)

**Pull request closed without merging**